code/game/objects/items/documents.dm

```diff
@@ -115,3 +229,220 @@
+
+/**
+ * # Bananium HONK-spect scanner
+ *
+ * An upgraded version of the fake N-spect scanner
+ *
+ * Can print things way faster, at full power the reports printed by this will destroy
+ * themselves and leave water behind when folding is attempted by someone who isn't an
+ * origami master. Printing at full power costs INSPECTOR_POWER_USAGE_HONK cell units
+ * instead of INSPECTOR_POWER_USAGE_NORMAL cell units.
+ */
+/obj/item/inspector/clown/bananium
+	name = "\improper Bananium HONK-spect scanner"
+	desc = "Honkmother-blessed inspection device. Performs inspections according to Clown protocols when activated, then \
+			prints a clowncrypted report regarding the maintenance of the station. Hard to replace."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "bananium_inspector"
+	w_class = WEIGHT_CLASS_SMALL
+	max_mode = BANANIUM_CLOWN_INSPECTOR_PRINT_SOUND_MODE_LAST
+	///How many more times can we print?
+	var/paper_charges = 150
+	///Max value of paper_charges
+	var/max_paper_charges = 400
+	///How much charges are restored per paper consumed
+	var/charges_per_paper = 16
+
+/obj/item/inspector/clown/bananium/proc/check_settings_legality()
+	if(print_sound_mode == INSPECTOR_PRINT_SOUND_MODE_NORMAL && time_mode == INSPECTOR_TIME_MODE_HONK)
+		if(cell.use(power_to_speak))
+			say("Setting combination forbidden by Geneva convention revision CCXXIII selected, reverting to defaults")
+		time_mode = INSPECTOR_TIME_MODE_SLOW
+		print_sound_mode = INSPECTOR_PRINT_SOUND_MODE_NORMAL
+		power_per_print = INSPECTOR_POWER_USAGE_NORMAL
+
+/obj/item/inspector/clown/bananium/screwdriver_act(mob/living/user, obj/item/tool)
+	. = ..()
+	check_settings_legality()
+
+/obj/item/inspector/clown/bananium/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(cell_cover_open)
+		check_settings_legality()
+	if(istype(I, /obj/item/paper/fake_report) || paper_charges >= max_paper_charges)
+		to_chat(user, "<span class='info'>\The [src] refuses to consume \the [I]!</span>")
+		return
+	if(istype(I, /obj/item/paper))
+		to_chat(user, "<span class='info'>\The [src] consumes \the [I]!</span>")
+		paper_charges = min(paper_charges + charges_per_paper, max_paper_charges)
+		qdel(I)
+
+/obj/item/inspector/clown/bananium/Initialize()
+	. = ..()
+	playsound(src, 'sound/effects/angryboat.ogg', 150, FALSE)
+
+/obj/item/inspector/clown/bananium/create_slip()
+	if(time_mode == INSPECTOR_TIME_MODE_HONK)
+		var/obj/item/paper/fake_report/water/slip = new(get_turf(src))
+		slip.generate_report(get_area(src))
+		return
+	return ..()
+
+/obj/item/inspector/clown/bananium/print_report(mob/user)
+	if(time_mode != INSPECTOR_TIME_MODE_HONK)
+		..()
+	else if(paper_charges == 0)
+		if(cell.use(power_to_speak))
+			say("ERROR! OUT OF PAPER! MAXIMUM PRINTING SPEED UNAVAIBLE! SWITCH TO A SLOWER SPEED TO OR PROVIDE PAPER!")
+		else
+			to_chat(user, "<span class='info'>\The [src] doesn't seem to be on... Perhaps it ran out of power?")
+	else
+		paper_charges--
+		..()
```

Comment:
```suggestion
/obj/item/inspector/clown/bananium/print_report(mob/user)
	if(time_mode != INSPECTOR_TIME_MODE_HONK)
		return ..()
	if(paper_charges == 0)
		if(cell.use(power_to_speak))
			say("ERROR! OUT OF PAPER! MAXIMUM PRINTING SPEED UNAVAIBLE! SWITCH TO A SLOWER SPEED TO OR PROVIDE PAPER!")
		else
			to_chat(user, "<span class='info'>\The [src] doesn't seem to be on... Perhaps it ran out of power?")
		return
	paper_charges--
	return ..()
```
* Preserve parent call return by default unless you have a reason not to. Even if it's null.
* Make use and abuse of early returns to reduce horizontal drift.